.gitmodules

```diff
@@ -10,3 +10,6 @@
   path = ccpp/physics
   url = https://github.com/NCAR/ccpp-physics
   branch = main
+[submodule "upp"]
+  path = upp
+  url = https://github.com/DusanJovic-NOAA/UPP
```

Comment:
@DusanJovic-NOAA When UPP being submodule in the authoritative fv3atm, how will an UPP version be checked out?from the latest branch "develop" or tag "XXX"?

Reply:
Submodules always point to specific commit, which can be a tag but it does not have to be. I guess most of the time, from the develop fv3atm branch we'll point to a commit in UPP develop branch. Production branch will probably point to a commit in UPP production branch.

Reply:
@DusanJovic-NOAA So in that case, I don't need create a tag from UPP repository site. We may check UPP version via hash number of upp directory on atmfv3 repository, right?

Reply:
Correct. fv3atm 'knows' only about UPP url (optionally branch) and commit hash.